test/xpu_api/numerics/complex.number/cmplx.over/conj.pass.cpp

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <complex>
+
+// template<class T>      complex<T>           conj(const complex<T>&);
+//                        complex<long double> conj(long double);
+//                        complex<double>      conj(double);
+// template<Integral T>   complex<double>      conj(T);
+//                        complex<float>       conj(float);
+
+#include "support/test_complex.h"
+
+#include "../cases.h"
+
+template <class T>
+void
+test(T x, typename std::enable_if<std::is_integral<T>::value>::type* = 0)
+{
+    static_assert((std::is_same<decltype(dpl::conj(x)), dpl::complex<double> >::value), "");
+    assert(dpl::conj(x) == dpl::conj(dpl::complex<double>(x, 0)));
+}
+
+template <class T>
+void
+test(T x, typename std::enable_if<std::is_floating_point<T>::value>::type* = 0)
+{
+    static_assert((std::is_same<decltype(dpl::conj(x)), dpl::complex<T>>::value), "");
+    assert(dpl::conj(x) == dpl::conj(dpl::complex<T>(x, 0)));
+}
+
+template <class T>
+void
+test(T x, typename std::enable_if<!std::is_integral<T>::value &&
+                                  !std::is_floating_point<T>::value>::type* = 0)
+{
+    static_assert((std::is_same<decltype(dpl::conj(x)), dpl::complex<T>>::value), "");
+    assert(dpl::conj(x) == dpl::conj(dpl::complex<T>(x, 0)));
+}
+
+template <class T>
+void
+test()
+{
+    test<T>(0);
+    test<T>(1);
+    test<T>(10);
+}
+
+ONEDPL_TEST_NUM_MAIN
+{
+    test<float>();
+    IF_DOUBLE_SUPPORT_IN_RUNTIME(test<double>())
+    IF_LONG_DOUBLE_SUPPORT_IN_COMPILETIME(test<long double>())
+    IF_DOUBLE_SUPPORT_IN_RUNTIME(test<int>())
```

Comment:
It very confuses me - the  macro tells about double support, but launched test parametrized bu int type. Such lines should be commented I guess.

Reply:
Added additional comments.